mantidproject/mantidimaging
# Common progress reporting interface

Fixes #106

Adds a progress reporting mechanism that is somewhat based on `Progress` in Mantid.

Anything that uses it has an optional `progress` parameter which may take a pre-configured instance (this is intended for the GUI when progress dialogs will be shown) or if left as the default will print an ASCII progress bar to standard out.

To test:
 - See that nothing is broken, pretty much all of the touched functionality is unit tested and they would have failed if the progress reporting was broken
- Load a large dataset (either on command line or GUI) and see  that the progress bar is shown and the reporting numbers look reasonable (i.e. correct number of images, bar progression looks right for progress). IMAT data from the last cycle should be good enough for this.
